Electron/Core/RoveProtocol/Rovecomm.ts

```diff
@@ -4,40 +4,62 @@ const dgram = require("dgram")
 const net = require("net")
 const EventEmitter = require("events")
 
+enum DataTypes {
+  INT8_T = 0,
+  UINT8_T = 1,
+  INT16_T = 2,
+  UINT16_T = 3,
+  INT32_T = 4,
+  UINT32_T = 5,
+  FLOAT_T = 6,
+}
+
 function decodePacket(
-  size: number,
+  dataType: number,
   dataLength: number,
   data: Buffer
 ): number[] {
-  const retArray = []
-  let readBytes
-
-  switch (size) {
-    case 1:
-      readBytes = function (i: number) {
-        return data.readUInt8(i)
-      }
+  let readBytes: (i: number) => number
+  let size: number
+  switch (dataType) {
```

Comment:
Shouldn't this switch on DataTypes[dataType]?

Reply:
I don't believe so.. At least not the way we have the rest of the switch set up currently. From Rovecomm Manifest, dataType is the number from the enum, such as 6 representing float. So we should switch on the number 6, and figure out which enum in DataTypes = 6. Correct?